mathesar/models.py

```diff
@@ -81,7 +81,9 @@ class DataFile(BaseModel):
         upload_to=model_utils.user_directory_path,
         validators=[FileExtensionValidator(allowed_extensions=['csv'])]
     )
-    table_imported_to = models.ForeignKey(Table, blank=True, null=True, on_delete=models.SET_NULL)
+    table_imported_to = models.ForeignKey(Table, related_name='data_file',
+                                          blank=True, null=True,
+                                          on_delete=models.SET_NULL)
     schema = models.ForeignKey(Schema, blank=True, null=True, on_delete=models.SET_NULL)
     user = models.ForeignKey(User, blank=True, null=True, on_delete=models.CASCADE)
 
```

Comment:
For some reason, I can't comment on the actual line I want to, but you don't need the `save` method here at all if you remove the `schema` from this model.